### PR TITLE
gh-94438: in frameobject's mark_stacks switch, the PUSH_EXC_INFO and POP_EXCEPT cases are no longer reachable

### PR DESCRIPTION
**Pull request by iritkatriel**
* Issue: gh-94438

### CODE REVIEW

File: Objects/frameobject.c [hunk 1]
```diff
@@ -304,10 +300,12 @@ mark_stacks(PyCodeObject *code_obj, int len)
...
-                    next_stack = push_value(next_stack, Except);
-                    next_stack = push_value(next_stack, Except);
-                    next_stack = push_value(next_stack, Except);
-                    stacks[i+1] = next_stack;
+                case POP_EXCEPT:
+                    /* These instructions only appear in exception handlers, which
+                     * skip this switch ever since the move to zero-cost exceptions
+                     * (their stack remains UNINITIALIZED because nothing sets it).
+                     */
+                    Py_UNREACHABLE();
```
Comment (mdboom) [inline Objects/frameobject.c:311]: Maybe this is overly pedantic, but would a `break` here make it clearer that there is no fall-through?

Comment (tiran) [reply Objects/frameobject.c:311]: +1

This would also help with static code analyzers that do not understand the unreachable macro.

Comment (iritkatriel) [reply Objects/frameobject.c:311]: Indeed.

**Comment by sweeneyde** [comment]
Maybe this is already understood, but stacks remaining `UNINITIALIZED` seems to be the cause of https://github.com/python/cpython/issues/92228

**Review by tiran — APPROVED** [review]
LGTM

**Review by pablogsal — APPROVED** [review]

**Comment by iritkatriel** [comment]
> Maybe this is already understood, but stacks remaining `UNINITIALIZED` seems to be the cause of #92228

I wasn't aware of this, I'll have a look thanks.

**Pull request merged**